File: frontend/src/types/portfolio.ts
/**
 * Combined portfolio summary from GET /portfolio/summary
 */

export type AssetClassSummary = {
  count: number;
  total_cost_basis: string;
  total_market_value: string;
  total_gain_loss: string;
  portfolio_return_percentage: string;
};

export type CombinedPortfolioSummary = {
  cash_balance: string;
  stocks: AssetClassSummary;
  treasuries: AssetClassSummary;
  total_cost_basis: string;
  total_market_value: string;
  total_gain_loss: string;
  portfolio_return_percentage: string;
};

File: backend/app/schemas/portfolio.py
"""Combined portfolio summary across asset classes."""

from decimal import Decimal

from pydantic import BaseModel


class AssetClassSummary(BaseModel):
    count: int
    total_cost_basis: Decimal
    total_market_value: Decimal
    total_gain_loss: Decimal
    portfolio_return_percentage: Decimal


class CombinedPortfolioSummaryResponse(BaseModel):
    cash_balance: Decimal
    stocks: AssetClassSummary
    treasuries: AssetClassSummary
    total_cost_basis: Decimal
    total_market_value: Decimal
    total_gain_loss: Decimal
    portfolio_return_percentage: Decimal
